src/mlpack/methods/ann/layer/instance_norm.hpp

```diff
@@ -0,0 +1,214 @@
+/**
+ * @file methods/ann/layer/instance_norm.hpp
+ * @author Anjishnu Mukherjee
+ *
+ * Definition of the Instance Normalization layer class.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_ANN_LAYER_INSTANCENORM_HPP
+#define MLPACK_METHODS_ANN_LAYER_INSTANCENORM_HPP
+
+#include <mlpack/prereqs.hpp>
+#include "layer_types.hpp"
+
+namespace mlpack {
+namespace ann /** Artificial Neural Network. */ {
+
+/**
+ * Declaration of the Instance Normalization layer class. The layer transforms
+ * the input data into zero mean and unit variance and then scales and shifts
+ * the data by parameters, gamma and beta respectively. These parameters are
+ * learnt by the network. The mean and standard-deviation are calculated
+ * per-dimension separately for each object in a mini-batch.
+ *
+ * If deterministic is false (training), the mean and variance are calculated
+ * and the data is normalized. If it is set to true (testing) then
+ * the mean and variance accrued over the training set is used.
+ *
+ * For more information, refer to the following paper,
+ *
+ * @code
+ * @article{Ulyanov17,
+ *   author    = {Dmitry Ulyanov, Andrea Vedaldi and
+ *                Victor Lempitsky},
+ *   title     = {Instance Normalization:
+ *                The Missing Ingredient for Fast Stylization},
+ *   year      = {2017},
+ *   url       = {https://arxiv.org/abs/1607.08022}
+ * }
+ * @endcode
+ *
+ * @tparam InputDataType Type of the input data (arma::colvec, arma::mat,
+ *         arma::sp_mat or arma::cube).
+ * @tparam OutputDataType Type of the output data (arma::colvec, arma::mat,
+ *         arma::sp_mat or arma::cube).
+ */
+template <
+  typename InputDataType = arma::mat,
+  typename OutputDataType = arma::mat
+>
+class InstanceNorm
+{
+ public:
+  //! Create the InstanceNorm object.
+  InstanceNorm();
+
+  /**
+   * Create the InstanceNorm layer object with the specified parameters.
+   *
+   * @param size The number of input units / channels.
+   * @param numFunctions The number of columns in input.
+   * @param eps The epsilon added to variance to ensure numerical stability.
+   * @param average Boolean to determine whether cumulative average is used for
+   *                updating the parameters or momentum is used.
```

Comment:
```suggestion
   *                updating the parameters or momentum.
```

I think if we remove the `is used` at the end it has a better flow.